renku_notebooks/api/classes/server.py

```diff
@@ -96,145 +66,122 @@ def __init__(
-                "an environment variable called IMAGE_REGISTRY"
-            )
+        self._repositories: list[Repository] = repositories
+        self._git_providers: list[GitProvider] | None = None
+        self._has_configured_git_providers = False
 
     @property
-    def gl_project(self):
-        return self._user.get_renku_project(self.gl_project_name)
-
-    @property
-    def gl_project_path(self) -> Optional[str]:
-        # NOTE: This is case sensitive and will reflect exact lower/uppercase combination of letters
-        # that is used in Gitlab for the path of the project.
-        gl_project = self.gl_project
-        return gl_project.path if gl_project else None
+    def user(self) -> AnonymousUser | RegisteredUser:
+        """Getter for server's user."""
+        return self._user
 
     @property
-    def gl_project_url(self) -> Optional[str]:
-        gl_project = self.gl_project
-        return gl_project.http_url_to_repo if gl_project else None
+    def k8s_client(self) -> K8sClient:
+        """Return server's k8s client."""
+        return self._k8s_client
 
     @property
-    def repositories(self) -> List[Dict[str, str]]:
-        if self._repositories is None:
-            self._repositories = [
-                asdict(
-                    Repository(
-                        namespace=self.namespace,
-                        project=self.gl_project_path,
-                        branch=self.branch,
-                        commit_sha=self.commit_sha,
-                        url=self.gl_project_url,
-                    )
-                )
-            ]
+    def repositories(self) -> list[Repository]:
+        # Configure git repository providers based on matching URLs.
+        if not self._has_configured_git_providers:
+            for repo in self._repositories:
+                found_provider = None
+                for provider in self.git_providers:
+                    if urlparse(provider.url).netloc == urlparse(repo.url).netloc:
+                        found_provider = provider
+                        break
+                if found_provider is not None:
+                    repo.provider = found_provider.id
+            self._has_configured_git_providers = True
 
         return self._repositories
 
     @property
-    def server_name(self):
-        """Make the name that is used to identify a unique user session"""
-        return make_server_name(
-            self._user.safe_username,
-            self.namespace,
-            self.project,
-            self.branch,
-            self.commit_sha,
+    def server_url(self) -> str:
+        """The URL where a user can access their session."""
+        if type(self._user) is RegisteredUser:
+            return urljoin(
+                "https://" + config.sessions.ingress.host,
+                f"sessions/{self.server_name}",
+            )
+        return urljoin(
+            "https://" + config.sessions.ingress.host,
```

Comment:
use f-string

Reply:
Done in d29ed41420c6b89d0ccbb60911289e2659412cd6.